bundler/lib/dependabot/bundler/file_parser/gemfile_declaration_finder.rb

```diff
@@ -8,20 +8,20 @@ module Bundler
     class FileParser
       # Checks whether a dependency is declared in a Gemfile
       class GemfileDeclarationFinder
-        def initialize(dependency:, gemfile:)
-          @dependency = dependency
```

Comment:
@dependency is used by a Dependabot runner internally, and unfortunately removing it would break much of how Dependabot runs.

Reply:
Looks like `GemfileDeclarationFinder` is only used once from the file parser so might be ok changing this.

Reply:
I assume this code is now public with https://github.com/dependabot/dependabot-core/pull/5608? In that case, should be easier to identify the downstream impacts of this change...